src/rules/deprecationRule.ts

```diff
@@ -210,23 +207,24 @@ function getDeprecationFromDeclarations(declarations?: ts.Declaration[]): string
     if (declarations === undefined) {
         return undefined;
     }
-    let declaration: ts.Node;
-    for (declaration of declarations) {
-        if (isBindingElement(declaration)) {
-            declaration = getDeclarationOfBindingElement(declaration);
-        }
-        if (isVariableDeclaration(declaration)) {
-            declaration = declaration.parent;
-        }
-        if (isVariableDeclarationList(declaration)) {
-            declaration = declaration.parent;
-        }
-        const result = getDeprecationFromDeclaration(declaration);
-        if (result !== undefined) {
-            return result;
-        }
+
+    const deprecations = declarations
+        .map(declaration => getDeprecationFromDeclaration(getRootDeclaration(declaration)))
+        .filter(deprecation => deprecation !== undefined);
+    const allDeclarationsDeprecated =
+        deprecations.length > 0 && deprecations.length === declarations.length;
+    return allDeclarationsDeprecated ? deprecations.join(", ") : undefined;
```

Comment:
It looks like there aren't any new test cases that exercise the behavior of multiple overloads all being deprecated. We'll need at least one.

Reply:
While trying to write tests for this I discovered it's not working as expected :p will look into it this weekend